Assets/Scripts/Layers/Adapters/ChainOfResponsibility.cs

```diff
@@ -0,0 +1,134 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using UnityEngine;
+using UnityEngine.Events;
+using UnityEngine.Networking;
+
+namespace Netherlands3D.Twin
+{
+    public class ChainOfResponsibility : MonoBehaviour
+    {
+        [Header("Data type adapters")] [Space(5)]
+        [SerializeField] private ScriptableObject[] dataTypeAdapters;
+        private IDataTypeAdapter[] dataTypeAdapterInterfaces;
+        public UnityEvent OnAdapterFound = new();
+
+        [Header("Events invoked on failures")] [Space(5)]
+        public UnityEvent<string> CouldNotFindAdapter = new();
+        public UnityEvent<string> OnDownloadFailed = new();
+
+        private string targetUrl = "";
+        public string TargetUrl { get => targetUrl; set => targetUrl = value; }
```

Comment:
Shouldn't this be a private set? To prevent external forces from corrupting a run?

Reply:
And is it necessary that this is a field + property? As far as I can see in the code it is not used elsewhere outside of starting the chain. And passing as variables + method arguments better than having more state that can possibly be corrupted or changed along the way

Reply:
I'll add comments to explain the seperate steps for setting url, and starting the chain